solidity/contracts/modules/MultisigModule.sol

```diff
@@ -0,0 +1,335 @@
+     * @param validator The address of the validator.
+     * @param validatorCount The number of enrolled validators in the validator set.
+     * @param commitment A commitment to the validator set and threshold.
+     */
+    event ValidatorUnenrolled(
+        uint32 indexed domain,
+        address indexed validator,
+        uint256 validatorCount,
+        bytes32 commitment
+    );
+
+    /**
+     * @notice Emitted when the quorum threshold is set.
+     * @param domain The remote domain of the validator set.
+     * @param threshold The new quorum threshold.
+     * @param commitment A commitment to the validator set and threshold.
+     */
+    event ThresholdSet(
+        uint32 indexed domain,
+        uint256 threshold,
+        bytes32 commitment
+    );
+
+    // ============ Constructor ============
+
+    // solhint-disable-next-line no-empty-blocks
+    constructor() Ownable() {}
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Enrolls a validator into a validator set.
+     * @dev Reverts if `_validator` is already in the validator set.
+     * @param _domain The remote domain of the validator set.
+     * @param _validator The validator to add to the validator set.
+     */
+    function enrollValidator(uint32 _domain, address _validator)
+        external
+        onlyOwner
+    {
+        require(_validator != address(0), "zero address");
+        require(validatorSet[_domain].add(_validator), "already enrolled");
+        bytes32 _commitment = _updateCommitment(_domain);
+        emit ValidatorEnrolled(
+            _domain,
+            _validator,
+            validatorCount(_domain),
+            _commitment
+        );
+    }
+
+    /**
+     * @notice Unenrolls a validator from a validator set.
+     * @dev Reverts if `_validator` is not in the validator set.
+     * @param _domain The remote domain of the validator set.
+     * @param _validator The validator to remove from the validator set.
+     */
+    function unenrollValidator(uint32 _domain, address _validator)
+        external
+        onlyOwner
+    {
+        require(validatorSet[_domain].remove(_validator), "!enrolled");
+        uint256 _validatorCount = validatorCount(_domain);
+        require(
+            _validatorCount >= threshold[_domain],
+            "violates quorum threshold"
+        );
+        bytes32 _commitment = _updateCommitment(_domain);
+        emit ValidatorUnenrolled(
+            _domain,
+            _validator,
+            _validatorCount,
+            _commitment
+        );
+    }
+
+    /**
+     * @notice Sets the quorum threshold.
+     * @param _domain The remote domain of the validator set.
+     * @param _threshold The new quorum threshold.
+     */
+    function setThreshold(uint32 _domain, uint256 _threshold)
+        external
+        onlyOwner
+    {
+        require(
+            _threshold > 0 && _threshold <= validatorCount(_domain),
```

Comment:
realizing there's not really a way to "sunset" a remote chain because we can't set the threshold back to 0 and we can't ever unenroll a validator if it results in the validator set size being less than the threshold. I guess this is fine because we can always do something like:
1. set threshold to 1
2. unenroll all validators but 1
3. enroll `address(0xdead)` or something
4. finally unenroll the last "real" validator we kept in step 2

This would possibly be made easier by a single function to set the entire validator set / threshold like I suggested. But I'm also not certain this is anything worth fretting about

Reply:
Agreed this feels like a hack.

Alternatively, the following changes would work:
1. Validators can be removed even if that pushes validator count < threshold, and
2. Threshold can be set to zero iff validator count is zero
3. Commitments are cleared if threshold is set to zero

Wdyt? OTOH this risks accidental "bricking" of the ISM, which, if is owned by an ICA, would be a problem.